app/com/linkedin/drelephant/spark/fetchers/SparkRestClient.scala

```diff
@@ -89,6 +93,36 @@ class SparkRestClient(sparkConf: SparkConf) {
     }
   }
 
+  def fetchLogData(appId: String, attemptId: Option[String])(
+      implicit ec: ExecutionContext
+  ): Future[Option[SparkLogDerivedData]] = {
+    val appTarget = apiTarget.path(s"applications/${appId}")
+    logger.info(s"calling REST API at ${appTarget.getUri}")
+
+    val logStem = attemptId.map(id => s"${appId}_$id").getOrElse(appId)
+    async {
+      resource.managed { getApplicationLogs(appTarget) }.acquireAndGet { zis =>
+        var seen = mutable.Buffer[String]()
+        var entry: ZipEntry = null
+        do {
```

Comment:
Can you please add a comment to which explains the logic of this do...while block?